core/src/parser.rs

```diff
@@ -988,21 +989,41 @@ impl<'a> Parser<'a> {
     }
 
     #[inline]
+    fn regular_expression_statement(&mut self) -> Result<Statement> {
+        let expression = try!(self.regular_expression());
+        Ok(expression.into())
+    }
+
+    #[inline]
+    fn regular_expression(&mut self) -> Result<Expression> {
+        let expression = try!(self.tokenizer.read_regular_expression());
+        match peek!(self) {
```

Comment:
This peek also shouldn't be necessary. Expression parser will handle complex expressions.